src/data_hub/api.py

```diff
@@ -904,6 +904,9 @@ def get_dataframe(
 
     # remove notebook columns because they are not
     # shown in the UI as columns
+    if columns is None:
+        return None
+
```

Comment:
if a database has no columns, then it has no data. there's nothing we can do except return None